keras/callbacks.py

```diff
@@ -399,7 +399,13 @@ def on_epoch_end(self, epoch, logs=None):
             self.epochs_since_last_save = 0
             filepath = self.filepath.format(epoch=epoch, **logs)
             if self.save_best_only:
-                current = logs.get(self.monitor)
+                if isinstance(logs.get(self.monitor), float):
+                    current = logs.get(self.monitor)
+                else:
+                    if isinstance(logs.get(self.monitor)[-1], float):
+                        current = logs.get(self.monitor)[-1]
+                    else:
+                        current = logs.get(self.monitor)[-1][-1]
```

Comment:
What case does this cover? Also, here we have twice an unchecked assumptions that we are dealing with a list; all we know is that it isn't a float. Not all non-floats are lists.